### PR TITLE
shell/scss: Fix dark theme code background for new ssh host modal

### PR DESCRIPTION
**Pull request by MahmoudHamdy02**
When adding a new SSH host for the first time, cockpit shows a modal with a line of code to verify the fingerprint. The background color in dark theme makes it almost impossible to read.

I added a condition for dark theme in the `scss`, let me know if this needs any changes!

## Before:

image

## After:

image

### CODE REVIEW

**Comment by marusak** [comment]
I wonder, wasn't this already fixed in https://github.com/cockpit-project/cockpit/pull/18308 ?
Pretty sure this is the same cause as https://github.com/cockpit-project/cockpit/pull/18307

**Comment by garrett** [comment]
@marusak: I thought so too... but that was for the login page (which doesn't actually use PatternFly; it just looks like it), this is for when you're logged into Cockpit (which actually uses PatternFly).

However, this is a 1-off fix. It is probably a wider spread problem with code blocks in dark mode and should be investigated and fixed at a higher level (probably PF overrides) if this is the case.

**Review by KKoukiou — CHANGES_REQUESTED** [review]
@MahmoudHamdy02 @marusak @garrett I think the correct way is to start using https://www.patternfly.org/v4/components/code-block/ instead of the pre. THe codeblock component has a black theme variant.

**Comment by KKoukiou** [comment]
@MahmoudHamdy02 is this clear what is suggested here?

**Comment by MahmoudHamdy02** [comment]
@KKoukiou Yup! I've converted one `<pre>` into PatternFly's `<CodeBlock>`. It looks correct in both light & dark modes automatically. Take a look and tell me what you think.

I've only kept the border radius from the old scss. Let me know what else should change in the design.

I will change the other occurrences of `<pre>` once the PF looks good

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: pkg/shell/shell.scss
Comment (KKoukiou) [reply pkg/shell/shell.scss:239]: And please squash the commits.

Comment (MahmoudHamdy02) [reply pkg/shell/shell.scss]: I've removed the old scss and replaced all the `<pre>`s with `<CodeBlocks>`. All commits should also be squashed now. 

However, I'm not sure where the MotD page is within cockpit, so I couldn't check if it works as it should. Let me know your thoughts.

File: pkg/shell/hosts_dialog.jsx
Comment (KKoukiou) [inline pkg/shell/hosts_dialog.jsx:466]: You removed this and some other classes, but I see these are used in the tests. 

test/verify/check-shell-multi-machine:                self.assertEqual(b.text('#hostkey-fingerprint'), fp)
test/verify/check-shell-multi-machine:                self.assertTrue(b.val('#hostkey-fingerprint'))


I suggest you keep the classes for the tests to work. 

If these is CSS that you want to drop thats' assosiated with the class, we need to check each case, and drop it if no longer relevant and adjust the classname like here, so that's affecting only the non-changed files:

pkg/static/login.css:#hostkey-fingerprint {
pkg/static/login.html:          <pre id="hostkey-fingerprint"></pre>

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: pkg/systemd/overview-cards/motdCard.jsx
Comment (KKoukiou) [inline pkg/systemd/overview-cards/motdCard.jsx:121]: same here, don't remove id, as we use it for the tests.

File: pkg/systemd/overview-cards/motdCard.scss
Comment (KKoukiou) [inline pkg/systemd/overview-cards/motdCard.scss:3]: This rule was actually intended

You can see the motd in http://localhost:9090/system after you add some  content in /etc/motd file.

Comment (KKoukiou) [reply pkg/systemd/overview-cards/motdCard.scss:3]: I just texted a bit interactively. THe idea is that you want o keep the looks of it exactly how it was. This heavily regressed. 
Please check this by yourself, if you manage to adjust the CSS to achieve the expected looks/ 

I can suggest that the only that can be removed is font-size setting as codeblock default font size is also small.

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: pkg/shell/hosts_dialog.jsx
Comment (MahmoudHamdy02) [reply pkg/shell/hosts_dialog.jsx]: Will look out for this in the future :+1:

File: pkg/systemd/overview-cards/motdCard.scss
Comment (MahmoudHamdy02) [reply pkg/systemd/overview-cards/motdCard.scss:3]: When using `<CodeBlockCode>` directly, it looks identical to the old `<pre>`, let me know if this is OK.

**Comment by MahmoudHamdy02** [comment]
I've kept the class names. As for motd, when using `<CodeBlockCode>` directly, it looks identical to the old `<pre>`. Let me know if this is OK!

File: pkg/shell/shell.scss
Comment (KKoukiou) [inline pkg/shell/shell.scss:227]: There is definitely some UI change here, so I will ask @garrett for review. Personally I think it's fine to drop the override. 

previous:
Screenshot from 2023-03-21 10-42-28

now:
Screenshot from 2023-03-21 10-50-42

**Review by KKoukiou — CHANGES_REQUESTED** [review]

**Review by KKoukiou — APPROVED** [review]
From my side approved. Inline in the comments there are before after screenshots, for the relevant parts. Will need final ack from @garrett who is the designer.

**Comment by garrett** [comment]
So this is definitely an improvement. However, it would ideally also have a copy icon for the command someone is supposed to run.

Code block is specifically for multi-line code blocks. Right at the top of the page, the PatternFly website states "_A code block is a component that contains 2 or more lines of read-only code. The code in a code block can be copied to the clipboard_." @ https://www.patternfly.org/v4/components/code-block

It's _probably_ better to use a read-only copy block with the PatternFly utility class `pf-u-font-family-monospace`

- https://www.patternfly.org/v4/components/clipboard-copy
- https://www.patternfly.org/v4/utilities/text/

image

And here's what that would look like with monospace (I did an override with `font-family: var(--pf-global--FontFamily--redhat--monospace);`):

image

**Review by garrett — CHANGES_REQUESTED** [review]
Additionally, this should not be a danger alert, but a warning, as something COULD be wrong, but something isn't NECESSARILY wrong.

Here's the original mockup for it... Please note it was made for the login page and therefore isn't a modal.

login-key-warnings

Here's what everything else is cropped:

image

It would need more work to adapt it into what it has grown into (with the how-to-verify block (which is behind a popover in this mockup).

**Comment by garrett** [comment]
So to make it clear:

I think we should address the main issue with the entry (no dark mode) by using a read-only copy block with the PatternFly utility class `pf-u-font-family-monospace`. It should be for the entry only, not the hash. (Neither are actually multiple lines of code, so they should not use the CodeBlock widget.)

(The rest was to show that there are some other problems with this modal, but this issue and PR are more focused at fixing the dark mode, which is fine. We can deal with other issues in follow-ups.)

**Review by KKoukiou — COMMENTED** [review]
@MahmoudHamdy02 how about one more iteration where you change everything to use what @garrett suggested here? https://github.com/cockpit-project/cockpit/pull/18491#issuecomment-1477721135

it indeed looks much better like this.
Note: everything but for the motd.

**Comment by garrett** [comment]
What about the utility class to make the font monospaced though?

Add `pf-u-font-family-monospace` to the `className` in React.

https://www.patternfly.org/v4/utilities/text

You'd need to make sure the text utility CSS is imported near the top (at the imports section) of the toplevel file, however, if it isn't already, like this:

```scss
@import "@patternfly/patternfly/utilities/Text/text.scss";
```

It's currently _not_ in `shell.scss`, so it should be included under the other import (for select).

Importing a utility means that you don't have to write custom CSS for it and can therefore depend on PatternFly instead. And it should be in the toplevel as we should only import things once per page (so sub-pages that import things can cause things to be imported multiple times, so it's best to always check the toplevel SCSS file for each page of Cockpit. In this case, the "page" is the special shell one.)

**Comment by MahmoudHamdy02** [comment]
OK, switched to the utility class instead of the CSS, let me know if it's good now

**Review by KKoukiou — APPROVED** [review]

**Comment by KKoukiou** [comment]
@MahmoudHamdy02 removed the motd change from this PR - as it was changing the UI and our designer is on PTO so he can't approve it today.

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**
